diff --git a/plugins/fwupd/gs-fwupd-app.c b/plugins/fwupd/gs-fwupd-app.c
index ae07e11ac..c7ae3d261 100644
--- a/plugins/fwupd/gs-fwupd-app.c
+++ b/plugins/fwupd/gs-fwupd-app.c
@@ -52,8 +52,59 @@ gs_fwupd_app_set_is_locked (GsApp *app, gboolean is_locked)
 	gs_app_set_metadata_variant (app, "fwupd::IsLocked", tmp);
 }
 
+#if FWUPD_CHECK_VERSION(1, 8, 1)
+static gchar * /* (transfer full) */
+gs_fwupd_problem_to_string (FwupdClient *client,
+			    FwupdDevice *dev,
+			    FwupdDeviceProblem problem)
+{
+	if (problem == FWUPD_DEVICE_PROBLEM_SYSTEM_POWER_TOO_LOW) {
+		if (fwupd_client_get_battery_level (client) == FWUPD_BATTERY_LEVEL_INVALID ||
+		    fwupd_client_get_battery_threshold (client) == FWUPD_BATTERY_LEVEL_INVALID) {
+			/* TRANSLATORS: as in laptop battery power */
+			return g_strdup (_("System power is too low to perform the update"));
+		}
+		return g_strdup_printf (
+		    /* TRANSLATORS: as in laptop battery power */
+		    _("System power is too low to perform the update (%u%%, requires %u%%)"),
+		    fwupd_client_get_battery_level (client),
+		    fwupd_client_get_battery_threshold (client));
+	}
+	if (problem == FWUPD_DEVICE_PROBLEM_UNREACHABLE) {
+		/* TRANSLATORS: for example, a Bluetooth mouse that is in powersave mode */
+		return g_strdup (_("Device is unreachable, or out of wireless range"));
+	}
+	if (problem == FWUPD_DEVICE_PROBLEM_POWER_TOO_LOW) {
+		if (fwupd_device_get_battery_level (dev) == FWUPD_BATTERY_LEVEL_INVALID ||
+		    fwupd_device_get_battery_threshold (dev) == FWUPD_BATTERY_LEVEL_INVALID) {
+			/* TRANSLATORS: for example the batteries *inside* the Bluetooth mouse */
+			return g_strdup_printf (_("Device battery power is too low"));
+		}
+		/* TRANSLATORS: for example the batteries *inside* the Bluetooth mouse */
+		return g_strdup_printf (_("Device battery power is too low (%u%%, requires %u%%)"),
+				        fwupd_device_get_battery_level (dev),
+				        fwupd_device_get_battery_threshold (dev));
+	}
+	if (problem == FWUPD_DEVICE_PROBLEM_UPDATE_PENDING) {
+		/* TRANSLATORS: usually this is when we're waiting for a reboot */
+		return g_strdup (_("Device is waiting for the update to be applied"));
+	}
+	if (problem == FWUPD_DEVICE_PROBLEM_REQUIRE_AC_POWER) {
+		/* TRANSLATORS: as in, wired mains power for a laptop */
+		return g_strdup (_("Device requires AC power to be connected"));
+	}
+	if (problem == FWUPD_DEVICE_PROBLEM_LID_IS_CLOSED) {
+		/* TRANSLATORS: lid means "laptop top cover" */
+		return g_strdup (_("Device cannot be used while the lid is closed"));
+	}
+	return NULL;
+}
+#endif
+
 void
-gs_fwupd_app_set_from_device (GsApp *app, FwupdDevice *dev)
+gs_fwupd_app_set_from_device (GsApp *app,
+		FwupdClient *client,
+		FwupdDevice *dev)
 {
 	GPtrArray *guids;
 
@@ -61,11 +112,6 @@ gs_fwupd_app_set_from_device (GsApp *app, FwupdDevice *dev)
 	if (fwupd_device_has_flag (dev, FWUPD_DEVICE_FLAG_UPDATABLE))
 		gs_app_set_state (app, AS_APP_STATE_UPDATABLE_LIVE);
 
-	/* only can be applied in systemd-offline */
-	if (fwupd_device_has_flag (dev, FWUPD_DEVICE_FLAG_ONLY_OFFLINE))
-		gs_app_set_metadata (app, "fwupd::OnlyOffline", "");
-
-
 	/* reboot required to apply update */
 	if (fwupd_device_has_flag (dev, FWUPD_DEVICE_FLAG_NEEDS_REBOOT))
 		gs_app_add_quirk (app, GS_APP_QUIRK_NEEDS_REBOOT);
@@ -105,16 +151,38 @@ gs_fwupd_app_set_from_device (GsApp *app, FwupdDevice *dev)
 	}
 	if (fwupd_device_get_created (dev) != 0)
 		gs_app_set_install_date (app, fwupd_device_get_created (dev));
-	if (fwupd_device_get_description (dev) != NULL) {
-		g_autofree gchar *tmp = NULL;
-		tmp = as_markup_convert (fwupd_device_get_description (dev),
-					 AS_MARKUP_CONVERT_FORMAT_SIMPLE, NULL);
-		if (tmp != NULL)
-			gs_app_set_description (app, GS_APP_QUALITY_NORMAL, tmp);
+
+#if FWUPD_CHECK_VERSION(1, 8, 1)
+	if (fwupd_device_get_problems (dev) != FWUPD_DEVICE_PROBLEM_NONE) {
+		g_autoptr(GString) problems = g_string_new (NULL);
+		for (guint i = 0; i < sizeof (FwupdDeviceProblem) * 8; i++) {
+			FwupdDeviceProblem problem = 1ull << i;
+			g_autofree gchar *tmp = NULL;
+			if (!fwupd_device_has_problem (dev, problem))
+				continue;
+			tmp = gs_fwupd_problem_to_string (client, dev, problem);
+			if (tmp == NULL)
+				continue;
+			if (problems->len)
+				g_string_append_c (problems, '\n');
+			g_string_append (problems, tmp);
+		}
+		if (problems->len)
+			gs_app_set_metadata (app, "GnomeSoftware::problems", problems->str);
+		else
+			gs_app_set_metadata (app, "GnomeSoftware::problems", NULL);
+	} else {
+		gs_app_set_metadata (app, "GnomeSoftware::problems", NULL);
 	}
+#endif
 
 	/* needs action */
-	if (fwupd_device_has_flag (dev, FWUPD_DEVICE_FLAG_NEEDS_BOOTLOADER))
+	if (fwupd_device_has_flag (dev, FWUPD_DEVICE_FLAG_NEEDS_BOOTLOADER)
+#if FWUPD_CHECK_VERSION(1, 8, 1)
+	    || fwupd_device_has_flag (dev, FWUPD_DEVICE_FLAG_UPDATABLE_HIDDEN)
+	    || fwupd_device_get_problems (dev) != FWUPD_DEVICE_PROBLEM_NONE
+#endif
+      )
 		gs_app_add_quirk (app, GS_APP_QUIRK_NEEDS_USER_ACTION);
 	else
 		gs_app_remove_quirk (app, GS_APP_QUIRK_NEEDS_USER_ACTION);
@@ -124,7 +192,6 @@ static gchar *
 gs_fwupd_release_get_name (FwupdRelease *release)
 {
 	const gchar *name = fwupd_release_get_name (release);
-#if FWUPD_CHECK_VERSION(1,2,7)
 	GPtrArray *cats = fwupd_release_get_categories (release);
 
 	for (guint i = 0; i < cats->len; i++) {
@@ -132,62 +199,168 @@ gs_fwupd_release_get_name (FwupdRelease *release)
 		if (g_strcmp0 (cat, "X-Device") == 0) {
 			/* TRANSLATORS: a specific part of hardware,
 			 * the first %s is the device name, e.g. 'Unifying Receiver` */
-			return g_strdup_printf (_("%s Device"), name);
+			return g_strdup_printf (_("%s Device Update"), name);
 		}
 		if (g_strcmp0 (cat, "X-System") == 0) {
 			/* TRANSLATORS: the entire system, e.g. all internal devices,
 			 * the first %s is the device name, e.g. 'ThinkPad P50` */
-			return g_strdup_printf (_("%s System"), name);
+			return g_strdup_printf (_("%s System Update"), name);
 		}
 		if (g_strcmp0 (cat, "X-EmbeddedController") == 0) {
 			/* TRANSLATORS: the EC is typically the keyboard controller chip,
 			 * the first %s is the device name, e.g. 'ThinkPad P50` */
-			return g_strdup_printf (_("%s Embedded Controller"), name);
+			return g_strdup_printf (_("%s Embedded Controller Update"), name);
 		}
 		if (g_strcmp0 (cat, "X-ManagementEngine") == 0) {
 			/* TRANSLATORS: ME stands for Management Engine, the Intel AMT thing,
 			 * the first %s is the device name, e.g. 'ThinkPad P50` */
-			return g_strdup_printf (_("%s ME"), name);
+			return g_strdup_printf (_("%s ME Update"), name);
 		}
 		if (g_strcmp0 (cat, "X-CorporateManagementEngine") == 0) {
 			/* TRANSLATORS: ME stands for Management Engine (with Intel AMT),
 			 * where the first %s is the device name, e.g. 'ThinkPad P50` */
-			return g_strdup_printf (_("%s Corporate ME"), name);
+			return g_strdup_printf (_("%s Corporate ME Update"), name);
 		}
 		if (g_strcmp0 (cat, "X-ConsumerManagementEngine") == 0) {
 			/* TRANSLATORS: ME stands for Management Engine, where
 			 * the first %s is the device name, e.g. 'ThinkPad P50` */
-			return g_strdup_printf (_("%s Consumer ME"), name);
+			return g_strdup_printf (_("%s Consumer ME Update"), name);
 		}
 		if (g_strcmp0 (cat, "X-Controller") == 0) {
 			/* TRANSLATORS: the controller is a device that has other devices
 			 * plugged into it, for example ThunderBolt, FireWire or USB,
 			 * the first %s is the device name, e.g. 'Intel ThunderBolt` */
-			return g_strdup_printf (_("%s Controller"), name);
+			return g_strdup_printf (_("%s Controller Update"), name);
 		}
 		if (g_strcmp0 (cat, "X-ThunderboltController") == 0) {
 			/* TRANSLATORS: the Thunderbolt controller is a device that
 			 * has other high speed Thunderbolt devices plugged into it;
 			 * the first %s is the system name, e.g. 'ThinkPad P50` */
-			return g_strdup_printf (_("%s Thunderbolt Controller"), name);
+			return g_strdup_printf (_("%s Thunderbolt Controller Update"), name);
+		}
+		if (g_strcmp0 (cat, "X-CpuMicrocode") == 0) {
+			/* TRANSLATORS: the CPU microcode is firmware loaded onto the CPU
+			 * at system bootup */
+			return g_strdup_printf (_("%s CPU Microcode Update"), name);
+		}
+		if (g_strcmp0 (cat, "X-Configuration") == 0) {
+			/* TRANSLATORS: configuration refers to hardware state,
+			 * e.g. a security database or a default power value */
+			return g_strdup_printf (_("%s Configuration Update"), name);
+		}
+		if (g_strcmp0 (cat, "X-Battery") == 0) {
+			/* TRANSLATORS: battery refers to the system power source */
+			return g_strdup_printf (_("%s Battery Update"), name);
+		}
+		if (g_strcmp0 (cat, "X-Camera") == 0) {
+			/* TRANSLATORS: camera can refer to the laptop internal
+			 * camera in the bezel or external USB webcam */
+			return g_strdup_printf (_("%s Camera Update"), name);
+		}
+		if (g_strcmp0 (cat, "X-TPM") == 0) {
+			/* TRANSLATORS: TPM refers to a Trusted Platform Module */
+			return g_strdup_printf (_("%s TPM Update"), name);
+		}
+		if (g_strcmp0 (cat, "X-Touchpad") == 0) {
+			/* TRANSLATORS: TouchPad refers to a flat input device */
+			return g_strdup_printf (_("%s Touchpad Update"), name);
+		}
+		if (g_strcmp0 (cat, "X-Mouse") == 0) {
+			/* TRANSLATORS: Mouse refers to a handheld input device */
+			return g_strdup_printf (_("%s Mouse Update"), name);
+		}
+		if (g_strcmp0 (cat, "X-Keyboard") == 0) {
+			/* TRANSLATORS: Keyboard refers to an input device for typing */
+			return g_strdup_printf (_("%s Keyboard Update"), name);
+		}
+		if (g_strcmp0 (cat, "X-StorageController") == 0) {
+			/* TRANSLATORS: Storage Controller is typically a RAID or SAS adapter */
+			return g_strdup_printf (_("%s Storage Controller Update"), name);
+		}
+		if (g_strcmp0 (cat, "X-NetworkInterface") == 0) {
+			/* TRANSLATORS: Network Interface refers to the physical
+			 * PCI card, not the logical wired connection */
+			return g_strdup_printf (_("%s Network Interface Update"), name);
+		}
+		if (g_strcmp0 (cat, "X-VideoDisplay") == 0) {
+			/* TRANSLATORS: Video Display refers to the laptop internal display or
+			 * external monitor */
+			return g_strdup_printf (_("%s Display Update"), name);
+		}
+		if (g_strcmp0 (cat, "X-BaseboardManagementController") == 0) {
+			/* TRANSLATORS: BMC refers to baseboard management controller which
+			 * is the device that updates all the other firmware on the system */
+			return g_strdup_printf (_("%s BMC Update"), name);
+		}
+		if (g_strcmp0 (cat, "X-UsbReceiver") == 0) {
+			/* TRANSLATORS: Receiver refers to a radio device, e.g. a tiny Bluetooth
+			 * device that stays in the USB port so the wireless peripheral works */
+			return g_strdup_printf (_("%s USB Receiver Update"), name);
+		}
+		if (g_strcmp0 (cat, "X-Drive") == 0) {
+			/* TRANSLATORS: drive refers to a storage device, e.g. SATA disk */
+			return g_strdup_printf (_("%s Drive Update"), name);
+		}
+		if (g_strcmp0 (cat, "X-FlashDrive") == 0) {
+			/* TRANSLATORS: flash refers to solid state storage, e.g. UFS or eMMC */
+			return g_strdup_printf (_("%s Flash Drive Update"), name);
+		}
+		if (g_strcmp0 (cat, "X-SolidStateDrive") == 0) {
+			/* TRANSLATORS: SSD refers to a Solid State Drive, e.g. non-rotating
+			 * SATA or NVMe disk */
+			return g_strdup_printf (_("%s SSD Update"), name);
+		}
+		if (g_strcmp0 (cat, "X-Gpu") == 0) {
+			/* TRANSLATORS: GPU refers to a Graphics Processing Unit, e.g.
+			 * the "video card" */
+			return g_strdup_printf (_("%s GPU Update"), name);
+		}
+		if (g_strcmp0 (cat, "X-Dock") == 0) {
+			/* TRANSLATORS: Dock refers to the port replicator hardware laptops are
+			 * cradled in, or lowered onto */
+			return g_strdup_printf (_("%s Dock Update"), name);
+		}
+		if (g_strcmp0 (cat, "X-UsbDock") == 0) {
+			/* TRANSLATORS: Dock refers to the port replicator device connected
+			 * by plugging in a USB cable -- which may or may not also provide power */
+			return g_strdup_printf (_("%s USB Dock Update"), name);
 		}
 	}
-#endif
 
 	/* default fallback */
 	return g_strdup (name);
 }
 
+static AsUrgencyKind
+gs_fwupd_release_urgency_to_as_urgency_kind (FwupdReleaseUrgency urgency)
+{
+	switch (urgency) {
+	case FWUPD_RELEASE_URGENCY_LOW:
+		return AS_URGENCY_KIND_LOW;
+	case FWUPD_RELEASE_URGENCY_MEDIUM:
+		return AS_URGENCY_KIND_MEDIUM;
+	case FWUPD_RELEASE_URGENCY_HIGH:
+		return AS_URGENCY_KIND_HIGH;
+	case FWUPD_RELEASE_URGENCY_CRITICAL:
+		return AS_URGENCY_KIND_CRITICAL;
+	case FWUPD_RELEASE_URGENCY_UNKNOWN:
+	default:
+		return AS_URGENCY_KIND_UNKNOWN;
+	}
+}
+
 void
 gs_fwupd_app_set_from_release (GsApp *app, FwupdRelease *rel)
 {
+	GPtrArray *locations = fwupd_release_get_locations (rel);
+
 	if (fwupd_release_get_name (rel) != NULL) {
 		g_autofree gchar *tmp = gs_fwupd_release_get_name (rel);
 		gs_app_set_name (app, GS_APP_QUALITY_NORMAL, tmp);
 	}
 	if (fwupd_release_get_summary (rel) != NULL) {
 		gs_app_set_summary (app, GS_APP_QUALITY_NORMAL,
-				    fwupd_release_get_summary (rel));
+						fwupd_release_get_summary (rel));
 	}
 	if (fwupd_release_get_homepage (rel) != NULL) {
 		gs_app_set_url (app, AS_URL_KIND_HOMEPAGE,
@@ -203,19 +376,26 @@ gs_fwupd_app_set_from_release (GsApp *app, FwupdRelease *rel)
 		gs_app_set_license (app, GS_APP_QUALITY_NORMAL,
 				    fwupd_release_get_license (rel));
 	}
-	if (fwupd_release_get_uri (rel) != NULL) {
-		gs_app_set_origin_hostname (app,
-					    fwupd_release_get_uri (rel));
-		gs_fwupd_app_set_update_uri (app, fwupd_release_get_uri (rel));
+
+	if (locations->len > 0) {
+		const gchar *uri = g_ptr_array_index (locations, 0);
+		/* typically the first URI will be the main HTTP mirror, and we
+		 * don't have the capability to use an IPFS/IPNS URL anyway */
+		gs_app_set_origin_hostname (app, uri);
+		gs_fwupd_app_set_update_uri (app, uri);
 	}
+
 	if (fwupd_release_get_description (rel) != NULL) {
 		g_autofree gchar *tmp = NULL;
-		tmp = as_markup_convert (fwupd_release_get_description (rel),
-					 AS_MARKUP_CONVERT_FORMAT_SIMPLE, NULL);
+#if AS_CHECK_VERSION(1, 0, 0)
+		tmp = as_markup_convert (fwupd_release_get_description (rel), AS_MARKUP_KIND_TEXT, NULL);
+#else
+		tmp = as_markup_convert_simple (fwupd_release_get_description (rel), NULL);
+#endif
+
 		if (tmp != NULL)
 			gs_app_set_update_details (app, tmp);
 	}
-#if FWUPD_CHECK_VERSION(1,3,3)
 	if (fwupd_release_get_detach_image (rel) != NULL) {
 		g_autoptr(AsScreenshot) ss = as_screenshot_new ();
 		g_autoptr(AsImage) im = as_image_new ();
@@ -227,5 +407,6 @@ gs_fwupd_app_set_from_release (GsApp *app, FwupdRelease *rel)
 			as_screenshot_set_caption (ss, NULL, fwupd_release_get_detach_caption (rel));
 		gs_app_add_screenshot (app, ss);
 	}
-#endif
+
+	gs_app_set_update_urgency (app, gs_fwupd_release_urgency_to_as_urgency_kind (fwupd_release_get_urgency (rel)));
 }
diff --git a/plugins/fwupd/gs-fwupd-app.h b/plugins/fwupd/gs-fwupd-app.h
index fdb63e529..140d4a09e 100644
--- a/plugins/fwupd/gs-fwupd-app.h
+++ b/plugins/fwupd/gs-fwupd-app.h
@@ -23,6 +23,7 @@ void			 gs_fwupd_app_set_update_uri		(GsApp		*app,
 void			 gs_fwupd_app_set_is_locked		(GsApp		*app,
 								 gboolean	 is_locked);
 void			 gs_fwupd_app_set_from_device		(GsApp		*app,
+								 FwupdClient	*client,
 								 FwupdDevice	*dev);
 void			 gs_fwupd_app_set_from_release		(GsApp		*app,
 								 FwupdRelease	*rel);
diff --git a/plugins/fwupd/gs-plugin-fwupd.c b/plugins/fwupd/gs-plugin-fwupd.c
index 15e2b4b6b..d89a915c7 100644
--- a/plugins/fwupd/gs-plugin-fwupd.c
+++ b/plugins/fwupd/gs-plugin-fwupd.c
@@ -93,20 +93,9 @@ void
 gs_plugin_initialize (GsPlugin *plugin)
 {
 	GsPluginData *priv = gs_plugin_alloc_data (plugin, sizeof(GsPluginData));
-	g_autofree gchar *user_agent = NULL;
-	g_autoptr(SoupSession) soup_session = NULL;
 
 	priv->client = fwupd_client_new ();
 
-	/* use a custom user agent to provide the fwupd version */
-	user_agent = fwupd_build_user_agent (PACKAGE_NAME, PACKAGE_VERSION);
-	soup_session = soup_session_new_with_options (SOUP_SESSION_USER_AGENT, user_agent,
-						      SOUP_SESSION_TIMEOUT, 10,
-						      NULL);
-	soup_session_remove_feature_by_type (soup_session,
-					     SOUP_TYPE_CONTENT_DECODER);
-	gs_plugin_set_soup_session (plugin, soup_session);
-
 	/* set name of MetaInfo file */
 	gs_plugin_set_appstream_id (plugin, "org.gnome.Software.Plugin.Fwupd");
 }
@@ -263,6 +252,8 @@ gs_plugin_fwupd_new_app_from_device (GsPlugin *plugin, FwupdDevice *dev)
 	if (fwupd_release_get_appstream_id (rel) == NULL)
 		return NULL;
 
+	GsPluginData *priv = gs_plugin_get_data (plugin);
+
 	/* get from cache */
 	id = as_utils_unique_id_build (AS_APP_SCOPE_SYSTEM,
 				       AS_BUNDLE_KIND_UNKNOWN,
@@ -291,7 +282,7 @@ gs_plugin_fwupd_new_app_from_device (GsPlugin *plugin, FwupdDevice *dev)
 	as_icon_set_name (icon, "application-x-firmware");
 	gs_app_add_icon (app, icon);
 	gs_fwupd_app_set_from_release (app, rel);
-	gs_fwupd_app_set_from_device (app, dev);
+	gs_fwupd_app_set_from_device (app, priv->client, dev);
 
 	if (fwupd_release_get_appstream_id (rel) != NULL)
 		gs_app_set_id (app, fwupd_release_get_appstream_id (rel));
@@ -331,7 +322,6 @@ gs_plugin_fwupd_new_app_from_device_raw (GsPlugin *plugin, FwupdDevice *device)
 	gs_app_set_version (app, fwupd_device_get_version (device));
 	gs_app_set_name (app, GS_APP_QUALITY_LOWEST, fwupd_device_get_name (device));
 	gs_app_set_summary (app, GS_APP_QUALITY_LOWEST, fwupd_device_get_summary (device));
-	gs_app_set_description (app, GS_APP_QUALITY_LOWEST, fwupd_device_get_description (device));
 	gs_app_set_origin (app, fwupd_device_get_vendor (device));
 	gs_fwupd_app_set_device_id (app, fwupd_device_get_id (device));
 	gs_app_set_management_plugin (app, "fwupd");
@@ -357,6 +347,7 @@ static GsApp *
 gs_plugin_fwupd_new_app (GsPlugin *plugin, FwupdDevice *dev, GError **error)
 {
 	FwupdRelease *rel = fwupd_device_get_release_default (dev);
+  GPtrArray *locations = fwupd_release_get_locations (rel);
 	GPtrArray *checksums;
 	const gchar *update_uri;
 	g_autofree gchar *basename = NULL;
@@ -408,7 +399,12 @@ gs_plugin_fwupd_new_app (GsPlugin *plugin, FwupdDevice *dev, GError **error)
 			     gs_app_get_update_version (app));
 		return NULL;
 	}
-	update_uri = fwupd_release_get_uri (rel);
+
+	/* typically the first URI will be the main HTTP mirror, and we
+	 * don't have the capability to use an IPFS/IPNS URL anyway */
+	if (locations->len > 0)
+		update_uri = g_ptr_array_index (locations, 0);
+
 	if (update_uri == NULL) {
 		g_set_error (error,
 			     GS_PLUGIN_ERROR,
@@ -595,9 +591,11 @@ gs_plugin_add_updates (GsPlugin *plugin,
 				g_autofree gchar *desc = NULL;
 				if (fwupd_release_get_description (rel) == NULL)
 					continue;
-				desc = as_markup_convert (fwupd_release_get_description (rel),
-							  AS_MARKUP_CONVERT_FORMAT_SIMPLE,
-							  NULL);
+#if AS_CHECK_VERSION(1, 0, 0)
+				desc = as_markup_convert (fwupd_release_get_description (rel), AS_MARKUP_KIND_TEXT, NULL);
+#else
+				desc = as_markup_convert_simple (fwupd_release_get_description (rel), NULL);
+#endif
 				if (desc == NULL)
 					continue;
 				g_string_append_printf (update_desc,
@@ -745,7 +743,13 @@ gs_plugin_refresh (GsPlugin *plugin,
 		return FALSE;
 	for (guint i = 0; i < remotes->len; i++) {
 		FwupdRemote *remote = g_ptr_array_index (remotes, i);
-		if (!fwupd_remote_get_enabled (remote))
+		gboolean is_enabled;
+#if FWUPD_CHECK_VERSION(1, 9, 4)
+		is_enabled = fwupd_remote_has_flag (remote, FWUPD_REMOTE_FLAG_ENABLED);
+#else
+		is_enabled = fwupd_remote_get_enabled (remote);
+#endif
+		if (!is_enabled)
 			continue;
 		if (fwupd_remote_get_kind (remote) == FWUPD_REMOTE_KIND_LOCAL)
 			continue;
@@ -799,10 +803,6 @@ gs_plugin_fwupd_install (GsPlugin *plugin,
 	/* set the last object */
 	g_set_object (&priv->app_current, app);
 
-	/* only offline supported */
-	if (gs_app_get_metadata_item (app, "fwupd::OnlyOffline") != NULL)
-		install_flags |= FWUPD_INSTALL_FLAG_OFFLINE;
-
 	gs_app_set_state (app, AS_APP_STATE_INSTALLING);
 	if (!fwupd_client_install (priv->client, device_id,
 				   filename, install_flags,
@@ -1049,7 +1049,14 @@ gs_plugin_add_sources (GsPlugin *plugin,
 		app = gs_app_new (id);
 		gs_app_set_kind (app, AS_APP_KIND_SOURCE);
 		gs_app_set_scope (app, AS_APP_SCOPE_SYSTEM);
-		gs_app_set_state (app, fwupd_remote_get_enabled (remote) ?
+
+		gboolean is_enabled;
+#if FWUPD_CHECK_VERSION(1, 9, 4)
+		is_enabled = fwupd_remote_has_flag (remote, FWUPD_REMOTE_FLAG_ENABLED);
+#else
+		is_enabled = fwupd_remote_get_enabled (remote);
+#endif
+		gs_app_set_state (app, is_enabled ?
 				  AS_APP_STATE_INSTALLED : AS_APP_STATE_AVAILABLE);
 		gs_app_add_quirk (app, GS_APP_QUIRK_NOT_LAUNCHABLE);
 		gs_app_set_name (app, GS_APP_QUALITY_LOWEST,
